test/loadEnvironment.test.ts

```diff
@@ -0,0 +1,104 @@
+import * as fs from "fs"
+import { loadEnvironment } from "../src/loadEnvironment"
+
+const getContentfulEnvironmentFileFactory = jest.fn((_type: string) => {
+  return () => ({
+    getContentTypes: () => [],
+    getLocales: () => [],
+  })
+})
+
+jest.mock(
+  require("path").resolve(process.cwd(), "./getContentfulEnvironment.js"),
+  () => getContentfulEnvironmentFileFactory("js"),
+  { virtual: true },
+)
+
+jest.mock(
+  require("path").resolve(process.cwd(), "./getContentfulEnvironment.ts"),
+  () => getContentfulEnvironmentFileFactory("ts"),
+  { virtual: true },
+)
+
+const tsNodeRegistererEnabled = jest.fn()
+const tsNodeRegister = jest.fn()
+
+jest.mock("ts-node", () => ({ register: tsNodeRegister }))
+
+describe("loadEnvironment", () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+    jest.restoreAllMocks()
+
+    getContentfulEnvironmentFileFactory.mockReturnValue(() => ({
+      getContentTypes: () => [],
+      getLocales: () => [],
+    }))
```

Comment:
This is to ensure that the mock implementation is present for each test, since I'm resetting and restoring all mocks between tests which is a best practice to ensure that you're not leaking implementations between tests that could result in unexpected behaviours (which could go unnoticed by causing tests to pass 😬).

Likewise setting up the mock on line 4 ensures that the mock object is valid for as much of its lifetime as possible.

Reply:
Makes sense! I'd love to avoid duplication if possible. What if we create a variable that contains the piece of duplicated code?

```ts
// line ~4
const contentfulEnvironment = () => ({
  getContentTypes: () => [],
  getLocales: () => [],
})
const getContentfulEnvironmentFileFactory = jest.fn((_type: string) => contentfulEnvironment)

// line ~33
getContentfulEnvironmentFileFactory.mockReturnValue(contentfulEnvironment)
```